worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionUtilities.java

```diff
@@ -0,0 +1,270 @@
+            final InputStream is = getInputStream(fv, dataStore);
+            streams.add(is);
+        }
+        return new SequenceInputStream(Collections.enumeration(streams));
+    }
+
+    public static void addDetectedLanguagesToDocument(final LanguageDetectorResult detectorResult, final Document document)
+    {
+        // Add DetectedLanguages_Status field to the document.
+        replaceDocumentField(
+            document,
+            LanguageDetectionConstants.Fields.DETECTED_LANGUAGES_STATUS,
+            detectorResult.getLanguageDetectorStatus().toString());
+
+        // Add DetectedLanguages_ReliableResult field to the document.
+        replaceDocumentField(
+            document,
+            LanguageDetectionConstants.Fields.DETECTED_LANGUAGES_RELIABLERESULT,
+            String.valueOf(detectorResult.isReliable()));
+
+        // For each language detected, add the name, language code and the percentage of the language detected within the text data to
+        // the document.
+        if (detectorResult.getLanguages() != null) {
+            int languageId = 0;
+            for (DetectedLanguage detectedLanguage : detectorResult.getLanguages()) {
+                languageId++;
+                replaceDocumentField(
+                    document,
+                    getLanguageNameFieldName(languageId),
+                    detectedLanguage.getLanguageName());
+
+                replaceDocumentField(
+                    document,
+                    getLanguageCodeFieldName(languageId),
+                    detectedLanguage.getLanguageCode());
+
+                replaceDocumentField(
+                    document,
+                    getLanguagePercentageFieldName(languageId),
+                    String.valueOf(detectedLanguage.getConfidencePercentage()));
+            }
+        }
+    }
+
+    public static void addDetectedLanguagesToDocument(final LanguageDetectorResult detectorResult, final Document document,
+                                                      final String fieldName)
+    {
+        final String field = getLanguageFieldName(fieldName);
+
+        if (document.getField(field).hasValues()) {
+            document.getField(field).clear();
+        }
+
+        boolean requiresUnknown = false;
+        // For each language detected, add the name, language code and the percentage of the language detected within the text data to
+        // the document.
+        if (detectorResult.getLanguages() != null) {
+            for (DetectedLanguage detectedLanguage : detectorResult.getLanguages()) {
+                if (!detectedLanguage.getLanguageCode().equals("un")) {
+                    document.getField(field).add(detectedLanguage.getConfidencePercentage()
+                        + "% " + detectedLanguage.getLanguageCode());
+                    requiresUnknown = true;
+                }
+            }
+            if (!requiresUnknown) {
+                document.getField(field).add("0% un");
```

Comment:
Should this be "100% un"?
Also, should it be added in the _detectorResult.getLanguages() == null_ case too?